code/modules/reagents/chemistry/reagents/water.dm

```diff
@@ -353,22 +353,48 @@
 			M.SetConfused(0)
 			return
 	if(ishuman(M) && M.mind && M.mind.vampire && !M.mind.vampire.get_ability(/datum/vampire_passive/full) && prob(80))
-		switch(current_cycle)
-			if(1 to 4)
-				to_chat(M, "<span class = 'warning'>Something sizzles in your veins!</span>")
-				M.mind.vampire.nullified = max(5, M.mind.vampire.nullified + 2)
-			if(5 to 12)
-				to_chat(M, "<span class = 'danger'>You feel an intense burning inside of you!</span>")
-				M.adjustFireLoss(1)
-				M.mind.vampire.nullified = max(5, M.mind.vampire.nullified + 2)
-			if(13 to INFINITY)
-				to_chat(M, "<span class = 'danger'>You suddenly ignite in a holy fire!</span>")
-				for(var/mob/O in viewers(M, null))
-					O.show_message(text("<span class = 'danger'>[] suddenly bursts into flames!<span>", M), 1)
-				M.fire_stacks = min(5,M.fire_stacks + 3)
-				M.IgniteMob()			//Only problem with igniting people is currently the commonly availible fire suits make you immune to being on fire
-				M.adjustFireLoss(3)		//Hence the other damages... ain't I a bastard?
-				M.mind.vampire.nullified = max(5, M.mind.vampire.nullified + 2)
+		var/mob/living/carbon/V = M
+		if(M.mind.vampire.bloodusable > 0)
+			M.Stuttering(1)
+			M.Jitter(30)
+			M.adjustStaminaLoss(5)
+			if(prob(20))
+				M.emote("scream")
+			if(M.mind.vampire.bloodusable > 20)
```

Comment:
This is really messy. Using `if` and `>` or `<` checks is generally something we shy away from these says. It's generally better to clamp values with `min`, `max`, or `Clamp`, where appropriate. It prevents needless checking and is a *lot* cleaner.